src/networking.c

```diff
@@ -2637,7 +2644,19 @@ void readQueryFromClient(connection *conn) {
         /* Read as much as possible from the socket to save read(2) system calls. */
         readlen = sdsavail(c->querybuf);
     }
+
+    ustime_t duration = 0;
+    if (io_threads_op == IO_THREADS_OP_IDLE) {
+        durationStartMonitor(duration);
+    }
+
     nread = connRead(c->conn, c->querybuf+qblen, readlen);
```

Comment:
This isn't a real representation of the IO READ, since on Linux the epoll will also execute part of the TCP stack to see if there is data available to read.

Reply:
Yes, but can we measure the time consumed in epoll? This may be inaccurate but at least it can provide a clear message when IO read is heavy, right?

Reply:
I don't know of a way to measure the epoll time.

Reply:
i think we should leave this out of scope.
we're trying to give some visibility on command latency by showing where the event loop spends time (before sending replies), and what portion of the event loop is consumed by each chore.
what Linux does inside epoll should be out of scope IMO.

Reply:
> i think we should leave this out of scope.

I very strongly disagree. I think we're scope creeping by adding this. 

> we're trying to give some visibility on command latency by showing where the event loop spends time (before sending replies), and what portion of the event loop is consumed by each chore.

We actually don't need to care for command latency. The real proxy of command latency is EL time and time from EL wakeup until command being written out to client.

Reply:
maybe i'm missing your point. you're arguing that since during the time redis's user space is sleeping on epoll, the kernel runs system space code (on our process time) that handles IO (TCP), and we should include that in our IO measurement?

i'm trying to add a breakdown of the composition of the event loop. i.e. we know how much time it took from wakeup to sleep, and i wanna know what where the components of that.. i.e. commands, IO, or other "cron" chores.

Reply:
>maybe i'm missing your point. you're arguing that since during the time redis's user space is sleeping on epoll, the kernel runs system space code (on our process time) that handles IO (TCP), and we should include that in our IO measurement?

I'm saying I don't care about this data, and don't believe in exposing useless metrics simply because we can compute them. I don't think the IO is particularly useful in aggregate. I think the one useful metric being exposed here is the one about total event loop time, which gives a nice proxy for command execution time.

Reply:
The IO read metrics should not be interesting until they unexpectedly are if the connection layer is unexpectedly slow. And since it's now extensible, I think the chances of that happening are not zero.